haystack/document_stores/weaviate.py

```diff
@@ -200,21 +209,22 @@ def __init__(
     def _get_auth_secret(
         username: Optional[str] = None,
         password: Optional[str] = None,
-        client_secret: Optional[str] = None,
-        access_token: Optional[str] = None,
-        expires_in: Optional[int] = 60,
-        refresh_token: Optional[str] = None,
+        api_key: Optional[str] = None,
         scope: Optional[str] = "offline_access",
     ) -> Optional[Union["AuthClientPassword", "AuthClientCredentials", "AuthBearerToken"]]:
-        if username and password:
+        if api_key:
+            return AuthApiKey(api_key=api_key)
+        elif username and password:
             return AuthClientPassword(username, password, scope=scope)
-        elif client_secret:
-            return AuthClientCredentials(client_secret, scope=scope)
-        elif access_token:
-            return AuthBearerToken(access_token, expires_in=expires_in, refresh_token=refresh_token)
-
         return None
 
+    @staticmethod
+    def _get_embedded_options(embedded_options: Optional[Dict[str, Any]] = None) -> weaviate.EmbeddedOptions:
+        if embedded_options is None:
+            return weaviate.EmbeddedOptions()
+        else:
+            return weaviate.EmbeddedOptions(**embedded_options)
```

Comment:
This can be simplified:
```suggestion
        embedded_options = embedded_options or {}
        return weaviate.EmbeddedOptions(**embedded_options)
```

Reply:
brilliant!